clcache.py

```diff
@@ -141,6 +144,18 @@ def getManifest(self, manifestHash):
             return None
 
 
+@contextlib.contextmanager
+def allSectionsLocked(repository):
+    sections = list(repository.sections())
```

Comment:
Okay cool, so this is about copying, not about using a different type. Then I'd write [copy.copy](https://docs.python.org/3/library/copy.html#copy.copy) to show your intent.